src/console_cmds.cpp

```diff
@@ -2095,6 +2097,154 @@ DEF_CONSOLE_CMD(ConFramerateWindow)
 	return true;
 }
 
+void ConDumpRoadTypes()
+{
+	IConsolePrintF(CC_DEFAULT, "  Flags:");
+	IConsolePrintF(CC_DEFAULT, "    c = catenary");
+	IConsolePrintF(CC_DEFAULT, "    l = no level crossings");
+	IConsolePrintF(CC_DEFAULT, "    X = no houses");
+	IConsolePrintF(CC_DEFAULT, "    h = hidden");
+	IConsolePrintF(CC_DEFAULT, "    T = buildable by towns");
+
+	extern RoadTypeInfo _roadtypes[ROADTYPE_END];
+	std::set<uint32> grfids;
+	for (RoadType rt = ROADTYPE_BEGIN; rt < ROADTYPE_END; rt++) {
+		const RoadTypeInfo &rti = _roadtypes[rt];
+		if (rti.label == 0) continue;
+		uint32 grfid = GetStringGRFID(rti.strings.name);
```

Comment:
Is there a specific intention/result when using the NewGRF of the name string?
I would have used "rti.grffile[ROTSG_GROUND]", that also works when NewGRF replace the sprites of the original tracktypes, without changing the name.

Reply:
Originally I looked for a way to get the source GRF of a spec/type and couldn't find one, so used the provider of the name string.
Happy to go with this instead.